src/v/storage/segment_utils.h

```diff
@@ -205,4 +205,20 @@ inline bool is_compactible(const model::record_batch& b) {
 offset_delta_time should_apply_delta_time_offset(
   ss::sharded<features::feature_table>& feature_table);
 
+template<typename Func, typename... Args>
```

Comment:
It looks like `Args` is not used. Should it be passed to `func` during `futurize_invoke`? It looks like right now the only two usages are `func(handle)` but perhaps using `Args` can make this helper more flexible.

Reply:
`Args` was meant to be removed. It's from a previous iteration. Will update.

Reply:
Updated